src/units/unit.hpp

```diff
@@ -1905,20 +1908,25 @@ class unit : public std::enable_shared_from_this<unit>
 		std::shared_ptr<const unit> parent;
 	};
 
-	recursion_guard update_variables_recursion() const;
-
-
-private:
+	recursion_guard update_variables_recursion(const config& ability) const;
 
 	const std::set<std::string> checking_tags_{"disable", "attacks", "damage", "chance_to_hit", "berserk", "swarm", "drains", "heal_on_hit", "plague", "slow", "petrifies", "firststrike", "poison", "damage_type"};
 	/**
-	 * Check if an ability is active.
+	 * Check if an ability is active. Use recursion checking for prevent crash in UI functions.
```

Comment:
"Includes checks to prevent excessive recursion."

"Use" is unclear whether it means "caller must use" or "includes", and it doesn't matter whether it's in a UI function or not.

Reply:
corrected

Reply:
Sorry for not being clear, I only meant replacing the second sentence. I'd like the first line to say:

    Check if an ability is active. Includes checks to prevent excessive recursion.

... and then I'm happy to merge it.

Reply:
it's done